index.d.ts

```diff
@@ -21,20 +21,27 @@ export interface Nodeback<E, R> {
   (err: E | null, value?: R): void
 }
 
-export interface ConcurrentFutureInstance<L, R> {
+export interface Functor<A> {
+  input: unknown
```

Comment:
I recommend changing `input` to a unique symbol, to prevent it from showing up in autocomplete:
```ts
declare const __functorInput: unique symbol;
export interface Functor<A> {
  [__functorInput]: unknown
  'fantasy-land/map'<B extends this[__functorInput]>(mapper: (value: A) => B): Functor<B>
}
```
You should also export the symbol if you want to support external `Functor` subtypes with `map`.

Reply:
Hey, I was thinking exactly this while going to sleep yesterday! I will make this change. Since it's a symbol I don't need it to have a obscure name, right?

Reply:
Another thought about the name: It's a placeholder for the yet-to-be-known type of `B` in `Functor<B>`, which is the *output* of the map operation, so calling it *input* may be misleading. But I'm guessing that when viewed from a different perspective, the name `input` does make sense?

Reply:
Actually this proposal doesn't seem to do the correct thing. Firstly, I need to use `[typeof __functorInput]`, because `__functorInput` is seen as a value, but when I do that, I guess the type becomes `unique symbol`, which is not specific enough, because we need the exact constant symbol that we've defined, as happens when using a constant string.

Reply:
It looks like there's no thing like a "symbol literal": https://www.staging-typescript.org/docs/handbook/literal-types.html

Reply:
Actually, my mistake. It does work with `typeof __functorInput`. I just forgot to change it in one occurrence.

Reply:
> to prevent it from showing up in autocomplete

It does have that effect. :)

> You should also export the symbol if you want to support external Functor subtypes with map.

Once I do, it starts showing up in autocomplete again, leading it to auto-import the symbol. :P

Reply:
`input` makes sense from the HKT standpoint, so that's why I chose it initially. `$T` works well though IMO.